src/wasm/wasm-type.cpp

```diff
@@ -897,7 +897,8 @@ bool Type::isFunction() const {
 
 bool Type::isData() const {
   if (isBasic()) {
-    return false;
+    // The only basic type that is considered data is a string.
+    return isString();
```

Comment:
Oh, if we already consider strings to be data, then there's some code above that we can simplify.

Reply:
Which code do you mean?

Reply:
The code I commented on in literal.cpp.